### DISCUSSION TITLE
What is "keep bluetooth connected" meant for ? Hot to use bluetooth and not wifi ? Is there a "switch off mammotion device" ?

### DISCUSSION THREAD

**Discussion by fschaer** [mikey0000/Mammotion-HA#173] [Q&A] [answered]
Hi,

I'm wondering what is the purpose of the integration option "keep bluetooth connected" ?
How can I know the integration is using BLE ? (I have setup a esphome bluetooth proxy on an esp32 plug from athom)
What should be preffered ? bluetooth or wifi ?

Could you elaborate a bit on wifi VS BLE support and features ?

Finally... I'm wondering if there would be some sort of "switch off mammotion device" remotely, or a way to tell it to go to sleep / release wheels breaks in order to prevent battery drain... I guess not, since it's not in the app, but still wondering... :]

Thanks

**Comment by AlohaFlyer** [comment]
I have the same question. What is the best use for the BLE in practice.

**Comment by mikey0000** [comment, marked as answer]
Sorry for not answering this sooner.

Wifi and BLE support the same features, keeping BLE connected is literally exactly that, do not disconnect which means you will not be able to use your phones BLE to connect to the mower until you disconnect HA.

Yes you can switch off the mower with an automation using the update on/off switch and error state.
There is no proper turn off remotely action.

BLE really is for people who install ESP32's inside their robot.

**Reply by gnoby** [reply]
> Yes you can switch off the mower with an automation using the update on/off switch and error state.

Thank you, I would love to be able to turn off the mower by using an automation. How can I do that?

**Reply by mikey0000** [reply]
There is a binary switch to turn updates off

**Comment by zerobit7** [comment]
Hi, I answer to this discussion becasue I think is partilly hte answer I'm looking.
I'd like to use this integration as "emergency controller", sometime my Luba looses positioning and the only way to make it working is to move it back and forth until it finds again its position. When I'm not home cannot do that, so I think with this integration I can do that using bluetooth.

However in the answer is writte "_BLE really is for people who install ESP32's inside their _robot"_
What does it mean? I alreay have an ESP32 as bluetooth proxy in my house for other things but not in the robot. I cannot use it for bluetooth connection? Btw I tried configuring with bluetooth but didn't work, with WiFi seems ok.

**Reply by mikey0000** [reply]
Problem is ble range, supposedly esp32-s3 with an antenna is pretty good

**Reply by zerobit7** [reply]
Well robot is far away from the antenna, at least 30meters. does not have an antenna, I might think to add an additional ESP32 for bluetooth in the garden as well
